actionpack/lib/action_controller/metal/request_forgery_protection.rb

```diff
@@ -311,9 +312,17 @@ def compare_with_real_token(token, session)
         ActiveSupport::SecurityUtils.secure_compare(token, real_csrf_token(session))
       end
 
+      # String compare raw CSRF values, both as a fast-path and for interop with other systems.
+      def raw_compare_with_real_token(raw_token, session)
+        ActiveSupport::SecurityUtils.secure_compare(raw_token, raw_csrf_token(session))
+      end
```

Comment:
Out of scope for this PR?

Reply:
> but not so much on the decoding side-- it's kind of irrelevant whether or not you can decode a previously encoded token as a particular format-- it just has to match.

That's distinct from this change, and it bears closer consideration of the implications. It's buried here alongside base64url encoding.

Reply:
Fair enough! I'll focus on the base64 stuff.